spec/requests/partners_spec.rb

```diff
@@ -0,0 +1,135 @@
+require "rails_helper"
+
+RSpec.describe "Partners", type: :request, include_shared: true do
+  let(:partner) { create(:partner) }
+  let(:user) { create(:user, partner: partner) }
+
+  context "when authenticated" do
+    before do
+      sign_in user
+    end
+    it "returns http success get#show" do
+      get partner_path(partner)
+
+      expect(response).to have_http_status(200)
+    end
+
+    it "should redirect to partner get#approve" do
+      stub_partner_requst
+      get partner_approve_path(partner)
+
+      expect(response).to redirect_to(partner)
+    end
+
+    it "does not allow access to partner post#index" do
+      get partners_path
+
+      expect(response).to have_http_status(:redirect)
+    end
+
+    describe "PATCH #update" do
+      subject { patch partner_path(partner, params: { partner: { name: "updated name" } }) }
```

Comment:
This subject definition seems a little odd to me. We still have to call `subject` in the examples to ensure that the request is executed before we can make expectations on the output. From one of the examples below...

```ruby
subject
expect(response).to redirect_to(partner)
```

As a developer reading this test file, this doesn't tell me much about what is being tested and requires me to seek the context elsewhere in the test file. My opinion is that this would read better to have the request inlined in the examples.